ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PreJoinRepartitionNode.java

```diff
@@ -143,25 +144,33 @@ private Optional<KeyFormat> maybeForceInternalKeyFormat(final KeyFormat format)
 
     if (requiresRepartition() || forceRepartition) {
       // Node is repartitioning already:
-      forcedInternalKeyFormat = Optional.of(format);
+      forcedInternalKeyFormat = Optional.of(removeSchemaIdIfPresent(format));
```

Comment:
Rather than applying the helper method in four different places in this method, it'd be better to apply it once upfront and then use the modified format throughout. That minimizes the risk that someone will add another usage of `format` into this method later and forget to apply the helper method (thereby introducing a bug).